packages/core/src/modules/connections/ConnectionsApi.ts

```diff
@@ -91,9 +91,14 @@ export class ConnectionsApi {
       imageUrl?: string
       protocol: HandshakeProtocol
       routing?: Routing
+      ourDid?: string
     }
   ) {
-    const { protocol, label, alias, imageUrl, autoAcceptConnection } = config
+    const { protocol, label, alias, imageUrl, autoAcceptConnection, ourDid } = config
+
+    if (ourDid && !config.routing) {
+      throw new AriesFrameworkError('If an external did is specified, routing configuration must be defined as well')
```

Comment:
This was set like that because, besides service information, routing object does also specify `mediatorId`, which is passed to `ConnectionService` when creating a connection. 

But I realized that we can infer the `mediatorId` by extracting recipient keys from the did document and querying mediation records by recipient keys, removing the need of using anything from `routing` parameter.